### ISSUE TITLE
[PinePhone]: adb remount failed

### ISSUE THREAD

**Issue by squidink7** [GloDroid/glodroid_manifest#161]
when I try to gain root access for adb, I need to remount the partitions, however when I run `adb remount` it results in this output:

```
Binder ioctl to enable oneway spam detection failed: Invalid argument
skip unmounted partition dev:/dev/block/dm-0 mnt:/system: Success
failed to remount partition dev:/dev/block/dm-1 mnt:/system_ext: Permission denied
failed to remount partition dev:/dev/block/dm-2 mnt:/vendor: Permission denied
failed to remount partition dev:/dev/block/dm-3 mnt:/product: Permission denied
remount failed
```
I tried running `adb root` and trying again but to no avail.

I'm using a pinephone and this happens both on v0.6.1 and v0.7.0

**Comment by squidink7** [comment]
Ok! Well best of luck. I wish I could help but unfortunately I'm far from experienced at Android development.
And thank you for your work on this project! being able to use my pinephone as an android development device is really awesome and actually got me into some basic app development! and that wouldn't have happened without the work you've been doing, so thank you again. You rock!

**Comment by KonstaT** [comment]
https://review.lineageos.org/c/LineageOS/android_system_core/+/318149 ?

**Comment by rsglobal** [comment]
@squidink7 ,

Thanks for your kind words.

@KonstaT ,

I believe this issue is related to some of kernel extra configurations or extra patches. This issue is only happens on kernel from Megous , which is used on the PinePhone, PineTab, Opi3. Other boards (rpi4, opi-plus2e, etc) has no issues with `adb remount`

**Comment by KonstaT** [comment]
Ok, just a FYI then that it's broken in current AOSP 12 branches (but fixed on master).

Can also reproduce that on my builds but only does affect remounting system partition.

**Comment by rsglobal** [comment]
I assume that lineage-rpi build is using system-as-root. (same as arpi).
GloDroid uses initrd as root.

**Comment by rsglobal** [comment]
Just checked with my local build (AOSP 12r2) on rpi4:

```
$ adb remount
Binder ioctl to enable oneway spam detection failed: Invalid argument
Waited one second for gsiservice (is service started? are binder threads started and available?)
Using overlayfs for /system
Using overlayfs for /system_ext
Using overlayfs for /vendor
Using overlayfs for /product
Now reboot your device for settings to take effect
remount succeeded
```

**Comment by KonstaT** [comment]
Everything Android 10 and up is system-as-root by default (https://source.android.com/devices/bootloader/partitions/system-as-root). I didn't realize GloDroid was any different but now I see you set `BOARD_BUILD_SYSTEM_ROOT_IMAGE := false` (https://github.com/GloDroid/glodroid_device/blob/master/common/boardconfig-common.mk#L64).

Android 12 I have running is just plain AOSP (android-12.0.0_r2 to be precise). Nothing LineageOS as it's probably not even buildable without bunch of gerrit patches and it doesn't have much over AOSP yet anyway.

**Comment by rsglobal** [comment]
IIRC, system-is-root was Android-10 requirements only. Starting from 11 initrd-as-root is required by VTS again. No?

**Comment by KonstaT** [comment]
IDK, is it? :o Android certification has never been any goal for me so I haven't looked into VTS (and last time I even tried to run CTS it was many many years ago on completely different device).

BTW, that document I linked is pretty confusing.
"Devices launching with Android 10:
...
Must set BOARD_BUILD_SYSTEM_ROOT_IMAGE to false."
AFAICT this is the retain the Android 9 behavior.

And later "In an Android 10 system-as-root configuration, the build always merges $TARGET_SYSTEM_OUT and $TARGET_ROOT_OUT into system.img; this config is the default behavior for all devices running Android 10.".
"

**Comment by rsglobal** [comment]
I found this:
https://cs.android.com/android/platform/superproject/+/master:build/make/core/config.mk;l=852?q=%22cannot%20be%20true%20for%20devices%20with%20dynamic%20partitions%22

So once you have dynamic partitions you can't use system-as-root.

**Comment by KonstaT** [comment]
Ah, you're using dynamic partitions as well (I'm not).

LOL, everything changes between Android versions so just try to keep up. :)

**Comment by rsglobal** [comment]
> LOL, everything changes between Android versions so just try to keep up. :)

https://www.youtube.com/watch?v=tjRln3WemG4&ab_channel=LinaroOrg

Here is a John's talk on this topic.

**Comment by KonstaT** [comment]
Yeah, actually already saw that earlier (watched your presentation as well :)).

Good points there that there isn't any actual reference point for maintaining an AOSP device configuration. You can get away using pretty legacy stuff (old HALs, old ways to implement stuff, etc) if it's not horribly broken. It just a matter how much "technical debt" you want to leave yourself trying to keep up.

**Comment by StuartRothrock** [comment]
I appreciate the hard work you guys have done with this. Any update? Is there anything a less than decent hacker can do to fix this? Thanks.

**Comment by rsglobal** [comment]
It should work in the next release, but next release will be based on Android master branch ( not 12 anymore but not 13 yet )

**Comment by StuartRothrock** [comment]
Thanks. Any sort of date for a working alpha or beta?

**Comment by rsglobal** [comment]
I can attach images next week or you can try building it from the sources (next branch)

**Comment by StuartRothrock** [comment]
Awesome. That'll work. I have a working build environment set up.

**Comment by squidink7** [comment]
If you could upload the built images here that would be really nice! I would build it myself but I don't have the disk space.

**Comment by StuartRothrock** [comment]
Will do. Glad to share and save others' efforts.

**Comment by rsglobal** [comment]
> Awesome. That'll work. I have a working build environment set up.

Looks I forgot to update next with my latest local version and you may have build issue related to vibrator. You may disable vibrator for now.

**Comment by StuartRothrock** [comment]
ok. I would also like to disable the disk/partition encryption. Would that
be an easy task to configure?

**Comment by rsglobal** [comment]
Yes, open fstab and remove fileencryprion=xxxxx from sunxi section.

**Comment by StuartRothrock** [comment]
If you mean /vendor/etc/fstab, I can't remount /dev/block/dm-2  or /vendor
with rw perms. I already tried changing it
in GloDroid/device/glodroid/platform/fstab/fstab.cpp and recompiling. It
never boots past the "android" splash. I'm willing to try something else.

**Comment by rsglobal** [comment]
> GloDroid/device/glodroid/platform/fstab/fstab.c

Yes, that should work. Make sure you erased user data partition.

**Comment by StuartRothrock** [comment]
It worked. I changed the fstab again and this time left off the comments I
added the last time to show my changes. Thanks for the suggestion. I
probably would not have tried it again without your reinforcement.

**Comment by squidink7** [comment]
Do you still have this image around? If so could you upload it to some file sharing site and give a link? I'd love to try this out for myself
Thank you!

**Comment by rsglobal** [comment]
I'll try to push remount related fixes asap. 2 overlayfs related configs has to be removed from defconfig to make it work properly.

**Comment by StuartRothrock** [comment]
squidink7 - if you are referring to me, I can directly share the images.tar.gz with you. How do I pm you?

**Comment by squidink7** [comment]
You could email them to me, my address is squidinkrf@gmail.com. Alternatively Matrix would work, I'm @squidink7:matrix.org over there.

**Comment by StuartRothrock** [comment]
@rsglobal If you had the time to make the changes for the remount problem, I should have my new mobo Tuesday and can test after that.The sdcard sense switch was broken in the original mobo.

**Comment by rsglobal** [comment]
Please test these images and give some feedback: https://github.com/rsglobal/glodroid_manifest/releases/tag/v0.7.1

**Comment by squidink7** [comment]
I got a "request would exceed partition size" when flashing the recovery

**Comment by StuartRothrock** [comment]
It loads. What am I to test?

**Comment by StuartRothrock** [comment]
If it was the adb remount - it returns this - 
`Waited one second for gsiservice (is service started? are binder threads started and available?)`
then after maybe 60 seconds:
```
Using overlayfs for /system
Using overlayfs for /system_ext
Using overlayfs for /vendor
Using overlayfs for /product
Now reboot your device for settings to take effect
remount succeeded
```
And I can modify files in the /system directory!!

**Comment by StuartRothrock** [comment]
@rsglobal How can I apply these source changes into my already built source repo?

**Comment by StuartRothrock** [comment]
@rsglobal If I copy the source files from your link to the .repo/manifests directory  and then start with the "repo sync -cq" - should that build properly?

**Comment by rsglobal** [comment]
Fixed in v0.7.1. (Release will be published soon)
